### PR TITLE
Add Adt to SMIR

### PR DESCRIPTION
**Pull request by spastorino**
r? @oli-obk

### CODE REVIEW

File: compiler/rustc_smir/src/rustc_internal/mod.rs [hunk 1]
```diff
@@ -43,6 +47,18 @@ impl<'tcx> Tables<'tcx> {
         self.def_ids.push(did);
         stable_mir::CrateItem(id)
     }
+
+    pub fn adt_def(&mut self, did: DefId) -> stable_mir::ty::AdtDef {
```
Comment (spastorino) [inline compiler/rustc_smir/src/rustc_internal/mod.rs:51]: I don't like that this is exactly the same as `item_def_id`. I can change this to make it DRY but unsure how much this is going to stay like this. @oli-obk ?

Comment (spastorino) [reply compiler/rustc_smir/src/rustc_internal/mod.rs:51]: Fixed!

File: compiler/rustc_smir/src/rustc_smir/mod.rs [hunk 2]
```diff
@@ -94,7 +94,25 @@ impl<'tcx> Tables<'tcx> {
...
             },
-            ty::Adt(_, _) => todo!(),
+            ty::Adt(adt_def, substs) => TyKind::RigidTy(RigidTy::Adt(
+                rustc_internal::adt_def(adt_def.did()),
+                AdtSubsts(
+                    substs
+                        .iter()
+                        .map(|arg| match arg.unpack() {
+                            ty::GenericArgKind::Lifetime(_) => {
+                                todo!();
```
Comment (oli-obk) [inline compiler/rustc_smir/src/rustc_smir/mod.rs:104]: You could use the opaque representation instead of panicking

**Pull request merged**